File: Week_6/10828. 스택/스택.java
import java.util.*;

public class Main {
    public static void main(String args[]) {
      Stack<Integer> stack = new Stack<>();
      Scanner sc = new Scanner(System.in);
      
      int n = sc.nextInt();
      StringBuilder sb = new StringBuilder();
      
      while(n > 0){
          n--;
          String str = sc.next();
          
          if(str.equals("push")) {
              stack.push(sc.nextInt());
              continue;
          }
          if(str.equals("pop")) sb.append(stack.isEmpty() ? -1 : stack.pop());
          if(str.equals("size")) sb.append(stack.size());
          if(str.equals("empty")) sb.append(stack.isEmpty() ? 1 : 0);
          if(str.equals("top")) sb.append(stack.isEmpty() ? -1 : stack.peek());
          
          sb.append("\n");
      }
        System.out.println(sb);
    }
}

File: Week_6/10828. 스택/스택.cc
#include<stdio.h>
#include<string.h>

struct stack{
        int stack[10001]={0};
    }stack;

main()
{
    int n,num;
    int cnt=0;
    scanf("%d",&n);
    while(n--)
    {
        char s[10];
        scanf("%s",s);
       
        if(!strcmp(s,"push"))
        {
            scanf("%d",&num);
            stack.stack[cnt]=num;
            cnt++;
        }
        else if(!strcmp(s,"pop"))
        {
            if(cnt!=0)
            {
                cnt--;
                printf("%d\n",stack.stack[cnt]);
                stack.stack[cnt]=0;
            }
            else printf("-1\n");
        }
        else if(!strcmp(s,"size"))
        {
            printf("%d\n",cnt);
        }
        else if(!strcmp(s,"empty"))
        {
            if(cnt!=0) printf("0\n");
            else printf("1\n");
        }
        else if(!strcmp(s,"top"))
        {
            if(cnt!=0) printf("%d\n",stack.stack[cnt-1]);
            else printf("-1\n");
        }
    }
}